cmake/sdp.cmake

```diff
@@ -0,0 +1,41 @@
+#
+# Copyright (c) 2024 Nordic Semiconductor ASA
+#
+# SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+#
+
+function(sdp_assembly_generate)
+  set(hrt_dir ${PROJECT_SOURCE_DIR}/src/hrt)
+  set(hrt_src ${hrt_dir}/hrt.c)
+  set(hrt_msg "Generating ASM file for Hard Real Time files.")
+  set(hrt_opts -g0 -P -fno-ident -fno-verbose-asm)
+
+  # The external static library that we are linking with does not know
+  # how to build for this platform so we export all the flags used in
+  # this zephyr build to the external build system.
+  zephyr_get_compile_options_for_lang(C options)
+  zephyr_get_compile_definitions_for_lang(C defines)
+  zephyr_get_include_directories_for_lang(C includes)
+  zephyr_get_system_include_directories_for_lang(C sys_includes)
+  # Replace "-I" with "-isystem" to treat all Zephyr headers as system headers
+  # that do not trigger -Werror.
+  string(REPLACE "-I" "-isystem" includes "${includes}")
+
+  set(compiler_options ${defines} ${options} ${includes} ${sys_includes})
+
+  zephyr_include_directories(include)
+
+  add_custom_target(asm_gen
+    BYPRODUCTS ${hrt_dir}/hrt.s
+    COMMAND ${CMAKE_C_COMPILER} ${compiler_options} ${hrt_opts} -S ${hrt_src}
```

Comment:
Would it be possible to specify the output file name? Right now it is a default one, which is `hrt.S`, which means that the old version is overwritten when asm_gen is run. It would be better if it was saved as something like `hrt_new.S` so that it could be easily checked if it changed.

Reply:
There is no problem with this, but I would prefer to stay with the current solution. 
With creating a new one, we can overlook the new file on commit as an out-of-tree file. If a file that is currently in the repo changes, we can always see the changes, for example using git.

Reply:
I agree, but at the same time, if we overwrite the assembly instead of creating a new one, there is a possibility it will be committed when not intended because someone may forget to do `git restore`. Also, ninja asm_update was supposed to replace the asm file with the new one.
When it comes to checking changes, I meant during build time. If we left overwriting asm, I would have to either check it by using git in CMake or create an additional CMake "step" that will rename the old asm file to something else when performing ninja asm_check. The question is: do we want it this way? @masz-nordic

Reply:
Yes, you are right, I forgot the next step, which should be `ninja asm_update`. Sorry :)
I updated the script. I also added a fix so that the script does not depend on the filename. Now it takes into account all `.c` files in the `hrt` directory and compiles it to asm.